dashboard/app/models/unit_group.rb

```diff
@@ -479,14 +479,25 @@ def can_view_version?(user = nil)
   end
 
   # @param family_name [String] The family name for a course family.
+  # @param locale [String] User or request locale. Optional.
   # @return [UnitGroup] Returns the latest stable version in a course family.
-  def self.latest_stable_version(family_name)
+  def self.latest_stable_version(family_name, locale: 'en-us')
     return nil if family_name.blank?
 
-    all_courses.select do |course|
+    stable_course_versions = all_courses.select do |course|
       course.family_name == family_name &&
         course.published_state == Curriculum::SharedCourseConstants::PUBLISHED_STATE.stable
-    end.max_by(&:version_year)
+    end.sort_by(&:version_year).reverse
+
+    # Only select stable, supported UnitGroups (ignore supported locales if locale is an English-speaking locale).
+    locale_str = locale&.to_s
+    if locale_str&.start_with?('en')
+      stable_course_versions.first
+    else
+      stable_course_versions.find do |cv|
```

Comment:
what happens here if its does not find one? Should there be a default back to the latest stable?

Reply:
It would return `nil` like in `unit.rb`, I was trying to keep it consistent with that. Should just UnitGroup default back to latest stable or should both?

Reply:
What happens in unit.rb when the method returns nil? How is that handled? I think making sure that the experience is consistent is the goal so if you have that with this then I'm good

Reply:
A lot of the uses of `Unit.latest_stable_version` is to check if that version is equal to a given version, so if it were to return nil it would just cause the conditional to return false.

The experience will be consistent everywhere else `UnitGroup.latest_stable_version` is used currently since the only place that is now passing in the locale is in `course_version`'s `recommended?` function (where it's being updated in this PR). So all current behavior should remain the same other than the `recommended?` function.